src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/actions/manual/ManualTriggerAction.java

```diff
@@ -671,28 +672,31 @@ public String toString() {
          * Finds the highest and lowest approval value of the approval's type for the specified change.
          *
          * @param res the change.
+         * @param patchSetNumber the patch set number.
          * @return the highest and lowest value. Or 0,0 if there are no values.
          */
-        public HighLow getApprovals(JSONObject res) {
+        public HighLow getApprovals(JSONObject res, int patchSetNumber) {
             logger.trace("Get Approval: {} {}", type, res);
             int highValue = Integer.MIN_VALUE;
             int lowValue = Integer.MAX_VALUE;
             if (res.has("currentPatchSet")) {
                 logger.trace("Has currentPatchSet");
                 JSONObject patchSet = res.getJSONObject("currentPatchSet");
-                if (patchSet.has("approvals")) {
-                    JSONArray approvals = patchSet.getJSONArray("approvals");
-                    logger.trace("Approvals: ", approvals);
-                    for (Object o : approvals) {
-                        JSONObject ap = (JSONObject)o;
-                        if (type.equalsIgnoreCase(ap.optString("type"))) {
-                            logger.trace("A {}", type);
-                            try {
-                                int approval = Integer.parseInt(ap.getString("value"));
-                                highValue = Math.max(highValue, approval);
-                                lowValue = Math.min(lowValue, approval);
-                            } catch (NumberFormatException nfe) {
-                                logger.warn("Gerrit is bad at giving me Approval-numbers!", nfe);
+                if (patchSet.has("number") && patchSet.has("approvals")) {
```

Comment:
It's because approvals are related only to the latest patchset and when selecting "Include All Patchsets", outdated patchsets should not have filled Verify and Code-Review column.